### PR TITLE
Dependencies bump

### PR DESCRIPTION
**Pull request by MDikkii**

### CODE REVIEW

File: .plop-templates/mvi/{{pascalCase name}}Module.kt.hbs [hunk 1]
```diff
@@ -3,10 +3,10 @@ package {{ basePackage }}.{{camelCase name }}
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
-import dagger.hilt.android.components.ActivityComponent
+import dagger.hilt.components.SingletonComponent
 
 @Module
-@InstallIn(ActivityComponent::class)
+@InstallIn(SingletonComponent::class)
```
Comment (krzdabrowski) [inline .plop-templates/mvi/{{pascalCase name}}Module.kt.hbs:9]: Hmm, should be a case for each Module?
Hard to say for a template IMO.

File: app/build.gradle [hunk 2]
```diff
@@ -34,16 +34,23 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
+    buildFeatures {
+        compose true
+    }
+
+    composeOptions {
+        kotlinCompilerExtensionVersion '1.0.0-rc02'
```
Comment (krzdabrowski) [inline app/build.gradle:43]: Could be 1.0.0 now 🙂

File: build.gradle [hunk 3]
```diff
@@ -11,13 +11,13 @@ buildscript {
         }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.1'
-        classpath 'com.google.gms:google-services:4.3.3'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.21"
-        classpath "com.github.ben-manes:gradle-versions-plugin:0.21.0"
+        classpath 'com.android.tools.build:gradle:7.0.0-rc01'
```
Comment (krzdabrowski) [inline build.gradle:14]: Same here, can be stable now.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-all.zip
```
Comment (krzdabrowski) [inline gradle/wrapper/gradle-wrapper.properties:6]: And 7.1.1 (or 7.2-RC-1 if you want to try Version Catalogs with plugins 🙂 )

File: gradles/deps.gradle [hunk 5]
```diff
@@ -45,8 +45,26 @@ dependencies {
...
+    implementation 'androidx.compose.material:material:1.0.0-rc02'
+    // Material design icons
+    implementation 'androidx.compose.material:material-icons-core:1.0.0-rc02'
+    implementation 'androidx.compose.material:material-icons-extended:1.0.0-rc02'
+    // Integration with activities
+    implementation 'androidx.activity:activity-compose:1.3.0-rc02'
+    // Integration with ViewModels
+    implementation 'androidx.lifecycle:lifecycle-viewmodel-compose:1.0.0-alpha07'
+    // Integration with observables
+    implementation 'androidx.compose.runtime:runtime-livedata:1.0.0-rc02'
```
Comment (krzdabrowski) [inline gradles/deps.gradle:68]: Isn't it redundant for LiveData? Do we use it in that project?

File: gradles/deps.gradle [hunk 6]
```diff
@@ -45,8 +45,26 @@ dependencies {
...
+    implementation 'androidx.compose.ui:ui:1.0.0-rc02'
+    // Tooling support (Previews, etc.)
+    implementation 'androidx.compose.ui:ui-tooling:1.0.0-rc02'
+    // Foundation (Border, Background, Box, Image, Scroll, shapes, animations, etc.)
+    implementation 'androidx.compose.foundation:foundation:1.0.0-rc02'
+    // Material Design
+    implementation 'androidx.compose.material:material:1.0.0-rc02'
+    // Material design icons
+    implementation 'androidx.compose.material:material-icons-core:1.0.0-rc02'
+    implementation 'androidx.compose.material:material-icons-extended:1.0.0-rc02'
```
Comment (krzdabrowski) [inline gradles/deps.gradle:62]: Wondering about that dependency, AFAIR it weights quite a lot?

File: gradles/deps.gradle [hunk 7]
```diff
@@ -45,8 +45,26 @@ dependencies {
...
+    // Integration with activities
+    implementation 'androidx.activity:activity-compose:1.3.0-rc02'
+    // Integration with ViewModels
+    implementation 'androidx.lifecycle:lifecycle-viewmodel-compose:1.0.0-alpha07'
+    // Integration with observables
+    implementation 'androidx.compose.runtime:runtime-livedata:1.0.0-rc02'
+    implementation 'androidx.compose.runtime:runtime-rxjava2:1.0.0-rc02'
 }
```
Comment (krzdabrowski) [inline gradles/deps.gradle:70]: And maybe let's consider Compose Navigation dependency with Compose-Navigation-Hilt helper dependency? 🙂

File: gradles/lib.gradle [hunk 8]
```diff
@@ -29,6 +28,11 @@ android {
 
     buildFeatures {
         viewBinding true
+        compose true
+    }
+
+    composeOptions {
+        kotlinCompilerExtensionVersion '1.0.0-rc02'
```
Comment (krzdabrowski) [inline gradles/lib.gradle:35]: Can be 1.0.0 as well.

File: user/src/main/java/com/example/user/presentation/login/LoginModule.kt [hunk 9]
```diff
@@ -3,10 +3,10 @@ package com.example.user.presentation.login
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
-import dagger.hilt.android.components.ActivityComponent
+import dagger.hilt.components.SingletonComponent
 
 @Module
-@InstallIn(ActivityComponent::class)
+@InstallIn(SingletonComponent::class)
```
Comment (krzdabrowski) [inline user/src/main/java/com/example/user/presentation/login/LoginModule.kt:9]: I believe the best component for that is
`@InstallIn(ViewModelComponent::class)` ?

File: user/src/main/java/com/example/user/presentation/register/RegisterModule.kt [hunk 10]
```diff
@@ -3,11 +3,11 @@ package com.example.user.presentation.register
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
-import dagger.hilt.android.components.ActivityComponent
+import dagger.hilt.components.SingletonComponent
 
 @Module
-@InstallIn(ActivityComponent::class)
-class RegisterModule() {
+@InstallIn(SingletonComponent::class)
```
Comment (krzdabrowski) [inline user/src/main/java/com/example/user/presentation/register/RegisterModule.kt:9]: Applies to all

File: user/src/main/java/com/example/user/presentation/login/LoginModule.kt (hunk 9 above)
Comment (MDikkii) [reply user/src/main/java/com/example/user/presentation/login/LoginModule.kt:9]: Thanks! I will definitely check this out, didn't make proper reaserch as far as I see. I was surprised that `ActivityComponent` doesn't work anymore as we have only one `Activity`. But it is probably too hard for Dagger to reason about it.

File: gradles/deps.gradle (hunk 6 above)
Comment (MDikkii) [reply gradles/deps.gradle:62]: Yup, I've added all but as we mostly use our own designed icons we can surely delete extended part.

File: gradles/deps.gradle (hunk 5 above)
Comment (MDikkii) [reply gradles/deps.gradle:68]: We don't use it for compose right now, and as far as I can see we moved away from live data at all. I didn't remember that :D

File: .plop-templates/mvi/{{pascalCase name}}Module.kt.hbs (hunk 1 above)
Comment (MDikkii) [reply .plop-templates/mvi/{{pascalCase name}}Module.kt.hbs:9]: As you commented above - it should be `ViewModelComponent` scoped probably. The templates are for `ui feature` which should be specialised enough to make it that way. Thanks for the suggestion about `ViewModelComponent`.

**Review by krzdabrowski — APPROVED** [review]

**Pull request merged**